zerver/views/push_notifications.py

```diff
@@ -66,3 +67,25 @@ def remove_android_reg_id(
     validate_token(token, PushDeviceToken.GCM)
     remove_push_device_token(user_profile, token, PushDeviceToken.GCM)
     return json_success(request)
+
+
+@human_users_only
+@has_request_variables
+def send_test_push_notification_api(
+    request: HttpRequest, user_profile: UserProfile, token: str = REQ()
+) -> HttpResponse:
+    # If a token is specified in the request, the test notification is supposed to be sent
+    # to that device. If no token is provided, the test notification should be sent to
+    # all devices registered for the user.
+    if token != "":
+        try:
+            devices = [PushDeviceToken.objects.get(token=token, user=user_profile)]
+        except PushDeviceToken.DoesNotExist:
+            raise JsonableError(_("Token does not exist"))
+    else:
+        devices = list(PushDeviceToken.objects.filter(user=user_profile))
+
+    for device in devices:
+        send_test_push_notification(user_profile, device)
```

Comment:
`send_apple_push_notification` and `send_android_push_notification` both take lists of devices, and the content is the same within one type of device.  Is it possible to not iterate here, and instead let them do their own iteration (which will enable 7787fe3f4904060e43fec8f5b78aee63147a9220 to do its work)?

Reply:
Oh yeah, that's a good idea. Pushed a `[squash]` commit for that, can you take a look if it makes sense now? This will only matter on Cloud right now, since it doesn't use the bouncer. A server that uses the bouncer still has to make requests to it one by one per device - until we make this bouncer endpoint accept taking in a list of tokens from the server. Probably a useful improvement to make, but maybe as a follow-up issue after this is merged?

Reply:
Yeah, that adjustment to the bouncer is probably reasonable as a follow-up.

Reply:
Yeah, this was mostly about people using this on Zulip Cloud, not through the push bouncer.

The `[squash]` LGTM!